New package: KSVD v1.0.0

JuliaRegistrator opened the pull request:
- Registering package: KSVD
- Repository: https://github.com/RomeoV/KSVD.jl
- Created by: @RomeoV
- Version: v1.0.0
- Commit: e8e58b8ed093e5f2d5424179c1aabe5692379cee
- Reviewed by: @RomeoV
- Reference: https://github.com/RomeoV/KSVD.jl/commit/e8e58b8ed093e5f2d5424179c1aabe5692379cee#commitcomment-136706162
- Description: K-SVD in Julia
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd6239b3448d27754bdec2d7653fc976e387b77b2f7c1a7b4c2998e806ded5c91713c797e810d1dbe3ec5806f49d97d66e7e9e7355b9f2bf4403b4fecf85abc623305eb152497272e66783f987ec54b16c4a002e6ad0efca9d7d4d4ec6a4cf60f41e4695734066fed3c45ad8e36d55a3d91f279edfc7075bc6670a90a19efa30ca2124695354f0dc6946b62af280282617675972e5c57850d29e33d9d19b91fce82 -->

RomeoV commented:
[noblock]

## Justification for the naming.
The KSVD algorithm is a relatively well known algorithm, e.g. [with it's own Wikipedia page](https://en.wikipedia.org/wiki/K-SVD). Although certainly now known to all Julia users, I believe that to people familiar with the SVD algorithm, the naming KSVD should be clear.

There are also similarly named packages like [`TSVD.jl`](https://github.com/JuliaLinearAlgebra/TSVD.jl).
In fact, for TSVD one could reason to instead named it `TruncatedSVD`. However, in this case, `KSVD` literally stands for `k-SVD`, i.e. the `k` is not short for anything.

However, we can still also consider names like these:
- `KSVDIteration` (however imo not `IterativeKSVD`, as that implies there's a closed-form KSVD)
- `KSVDAlgorithm` (I think the `Algorithm` would be redundant.)
- `DictionaryLearningKSVD` (this would perhaps make sense if there was also `DictionaryLearningBase`, etc, but currently there is not and also I don't think anybody plans to make one)

Please let me know your thoughts.
Thanks!

JuliaRegistrator committed:
New package: KSVD v1.0.0

UUID: 778123d0-3295-4563-8004-fb5c576fbcd9
Repo: https://github.com/RomeoV/KSVD.jl.git
Tree: 1d65cf4ea936b74ff2943612fb2455351b318bed

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

DilumAluthge committed:
Merge branch 'master' into registrator-ksvd-778123d0-v1.0.0-4f5e76aba1

mkitti commented:
[noblock]

Thoughts from Slack:

> Would have been nicer to be a bit descriptive, but we already have things like TSVD. So should be ok.
> 
> It is a bit unfortunate, in that it is not an implementation of the SVD, but uses the SVD to implement clustering algorithms. So KSVDClustering or something might be a better name - and perhaps it is worth asking the authors to reconsider the name to see if we can do something better. If not, then I would go ahead and accept.

goerz commented:
At least "KSVD" in Google gives the "correct" hit as the first result, which I tend to use as a benchmark for whether acronyms are OK in package names. As long as this package is an "authoritative" implementation of k-SVD, I'd be okay with the name, although I'd definitely also consider `KSVDClustering`

ViralBShah commented:
Let's merge this.
